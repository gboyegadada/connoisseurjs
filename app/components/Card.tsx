import React, { ReactNode, FC } from 'react'
import { ResultItem } from '../types/search'
import { MdLocationCity, MdLocationOn } from 'react-icons/md'

interface CardProps {
    children?: ReactNode
    data: ResultItem,
    key: string
}

const Card: FC<CardProps> = (props) => {
    const { uniqueId, title, uri, raw } = props.data
    const numFormat = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'CAD' });

    return (
        <li className='card' key={uniqueId}>
            <a href={uri}>
                <img className='card-image' src={raw.tpthumbnailuri} />
            </a>
            <a href={uri}>
                <h5>{title}</h5>
            </a>
            <ul className='card-info'>
                <li><MdLocationOn /> {raw.tppays}</li>
                <li><MdLocationCity /> {raw.tpregion}</li>
                <li>
                { numFormat.format(parseFloat(raw.tpprixnum)) }
                </li>
            </ul>
            <ul className='tags'></ul>
        </li>
    )
}

export default Card