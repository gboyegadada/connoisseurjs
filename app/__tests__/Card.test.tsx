import React from 'react';
import { shallow } from 'enzyme'

import Card from "../components/Card";


describe("<Card />", () => {
  test("should display clickable item card", async () => {
    const main = shallow(<Card 
            key='-1$0$30292.23337$https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732'
            data={{ 
              "title": "Marc TempÃ© Gewurztraminer Mambourg grand cru 2009",
              "uri": "https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732",
              "uniqueId": "-1$0$30292.23337$https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732",
              "raw": {
                "tpregion": "Alsace",
                "tppays": "France",
                "tpthumbnailuri": "http://s7d9.scene7.com/is/image/SAQ/12184732_is",
                "tpprixnum": 62.25,
              }
            }}
        />)

    expect(main).toMatchSnapshot()
  });
});