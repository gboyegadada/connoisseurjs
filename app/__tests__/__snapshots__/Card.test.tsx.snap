// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should display clickable item card 1`] = `
<li
  className="card"
  key="-1$0$30292.23337$https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732"
>
  <a
    href="https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732"
  >
    <img
      className="card-image"
      src="http://s7d9.scene7.com/is/image/SAQ/12184732_is"
    />
  </a>
  <a
    href="https://www.saq.com/page/fr/saqcom/vin-blanc/marc-tempe-gewurztraminer-mambourg-grand-cru-2009/12184732"
  >
    <h5>
      Marc Temp√© Gewurztraminer Mambourg grand cru 2009
    </h5>
  </a>
  <ul
    className="card-info"
  >
    <li>
      <MdLocationOn />
       
      France
    </li>
    <li>
      <MdLocationCity />
       
      Alsace
    </li>
    <li>
      CA$62.25
    </li>
  </ul>
  <ul
    className="tags"
  />
</li>
`;
